dbms/src/TiDB/Schema/SchemaBuilder.cpp

```diff
@@ -373,6 +378,51 @@ void SchemaBuilder<Getter, NameMapper>::applySetTiFlashReplica(DatabaseID databa
     }
 }
 
+template <typename Getter, typename NameMapper>
+void SchemaBuilder<Getter, NameMapper>::applyPartitionAlter(DatabaseID database_id, TableID table_id)
+{
+    auto table_info = getter.getTableInfo(database_id, table_id);
+    if (table_info == nullptr) // the database maybe dropped
+    {
+        LOG_DEBUG(log, "table is not exist in TiKV, may have been dropped, table_id={}", table_id);
+        return;
+    }
+
+    table_id_map.emplaceTableID(table_id, database_id);
+    LOG_DEBUG(log, "register table to table_id_map, database_id={} table_id={}", database_id, table_id);
+
+    if (!table_info->isLogicalPartitionTable())
+    {
+        return;
+    }
+
+    // If table is partition table, we will create the logical table here.
+    // Because we get the table_info, so we can ensure new_db_info will not be nullptr.
+    auto new_db_info = getter.getDatabase(database_id);
+    applyCreateStorageInstance(new_db_info, table_info);
+
+    for (const auto & part_def : table_info->partition.definitions)
```

Comment:
when do applyPartitionDiff(new_db_info, table_info, storage), we also will emplacePartitionTableId based on the definitions, why do we also add it here?

Reply:
this function is called when
SchemaActionType::ActionAlterTablePartitioning / SchemaActionType::ActionRemovePartitioning with `diff.table_id != diff.old_table_id`.

when executing `alter table xxx partition by ...`, tiflash will first accept a schema diff with `ActionAlterTablePartitioning` with `diff.table_id == diff.old_table_id`. The new added partitions will be added by  `applyPartitionDiff`
Then tidb will create a new logical table add transfer the partitions to the new logical table. tiflash will accept a schema diff with `ActionAlterTablePartitioning` with `diff.table_id != diff.old_table_id`, then run into this function

Reply:
I get what you mean, let me check it

Reply:
Actually we create the Storage instance by `table_info` in line 402, then try to execute `applyPartitionDiff`. Then nothing must change between the `table_info` and `storage.table_info`. So calling `applyPartitionDiff` is redundant.
Then this function is simple the same as `applyCreateTable`. So I have remove the function `applyPartitionAlter `
https://github.com/pingcap/tiflash/blob/01a1860ee10b134f60c1cde2b5aabb75abaef397/dbms/src/TiDB/Schema/SchemaBuilder.cpp#L250-L262